src/frontend/src/optimizer/plan_visitor/execution_mode_decider.rs

```diff
@@ -42,6 +55,22 @@ impl PlanVisitor for ExecutionModeDecider {
     /// select * from t where k = 1
     /// select * from t where id between 1 and 5
     fn visit_batch_seq_scan(&mut self, batch_seq_scan: &BatchSeqScan) -> bool {
+        let reader = batch_seq_scan
+            .base
+            .ctx()
+            .session_ctx()
+            .env()
+            .catalog_reader()
+            .read_guard();
+        if let Some(table_stat) = reader
+            .table_stats()
+            .table_stats
+            .get(&batch_seq_scan.core().table_catalog.id().table_id())
```

Comment:
nit: Is it possible that there are multiple `Scan` node for the same table? Should we deduplicate them when estimating the scan bytes considering block cache?